helper/winsvc/service_windows.go

```diff
@@ -0,0 +1,41 @@
+//+build windows
+
+package winsvc
+
+import (
+	wsvc "golang.org/x/sys/windows/svc"
+)
+
+type serviceWindows struct{}
+
+func init() {
+	interactive, err := wsvc.IsAnInteractiveSession()
+	if err != nil {
+		panic(err)
+	}
+	// Cannot run as a service when running interactively
+	if interactive {
+		return
+	}
+	go func() {
+		_ = wsvc.Run("", serviceWindows{})
```

Comment:
[nitpick] Seems like this could be shortened to:
```suggestion
		go wsvc.Run("", serviceWindows{})
```

Reply:
Weird. With this fix we're getting the following on Appveyor:

```
# github.com/hashicorp/nomad/helper/winsvc
helper\winsvc\service_windows.go:20:35: cannot call non-function svc.Run("", serviceWindows literal) (type error)
Command exited with code 2
```

Reply:
Same locally:

```
▶ GOOS=windows go build -o bin/nomad.exe
# github.com/hashicorp/nomad/helper/winsvc
helper/winsvc/service_windows.go:20:35: cannot call non-function svc.Run("", serviceWindows literal) (type error)
```

This is calling https://github.com/golang/sys/blob/master/windows/svc/service.go#L309 which looks a heck of a lot like a function to me. 😕

Reply:
😊 

Nevermind, I'm a doofus and was calling the return value of the function instead of the function.